src/Features/Core/Portable/EmbeddedLanguages/StackFrame/StackFrameParser.cs

```diff
@@ -0,0 +1,455 @@
+            }
+
+            if (builder.Count == 0)
+            {
+                return ParseResult<StackFrameTypeArgumentList>.Abort;
+            }
+
+            if (closeToken.IsMissing)
+            {
+                return ParseResult<StackFrameTypeArgumentList>.Abort;
+            }
+
+            var separatedList = new EmbeddedSeparatedSyntaxNodeList<StackFrameKind, StackFrameNode, StackFrameIdentifierNameNode>(builder.ToImmutable());
+            return new(new(openToken, separatedList, closeToken));
+        }
+
+        /// <summary>
+        /// MyNamespace.MyClass.MyMethod[|(string s1, string s2, int i1)|]
+        /// Takes parameter declarations from method text and parses them into a <see cref="StackFrameParameterList"/>.
+        /// </summary>
+        private ParseResult<StackFrameParameterList> ParseMethodParameters()
+        {
+            if (!_lexer.ScanCurrentCharAsTokenIfMatch(StackFrameKind.OpenParenToken, scanTrailingWhitespace: true, out var openParen))
+            {
+                return ParseResult<StackFrameParameterList>.Abort;
+            }
+
+            if (_lexer.ScanCurrentCharAsTokenIfMatch(StackFrameKind.CloseParenToken, out var closeParen))
+            {
+                return new(new(openParen, EmbeddedSeparatedSyntaxNodeList<StackFrameKind, StackFrameNode, StackFrameParameterDeclarationNode>.Empty, closeParen));
+            }
+
+            using var _ = ArrayBuilder<StackFrameNodeOrToken>.GetInstance(out var builder);
+
+            while (true)
+            {
+                var (success, parameterNode) = ParseParameterNode();
+                if (!success)
+                {
+                    return ParseResult<StackFrameParameterList>.Abort;
+                }
+
+                RoslynDebug.AssertNotNull(parameterNode);
+                builder.Add(parameterNode);
+
+                if (!_lexer.ScanCurrentCharAsTokenIfMatch(StackFrameKind.CommaToken, out var commaToken))
+                {
+                    break;
+                }
+
+                builder.Add(commaToken);
+            }
+
+            if (!_lexer.ScanCurrentCharAsTokenIfMatch(StackFrameKind.CloseParenToken, out closeParen))
+            {
+                return ParseResult<StackFrameParameterList>.Empty;
+            }
+
+            var parameters = new EmbeddedSeparatedSyntaxNodeList<StackFrameKind, StackFrameNode, StackFrameParameterDeclarationNode>(builder.ToImmutable());
+            return new(new(openParen, parameters, closeParen));
+        }
+
+        /// <summary>
+        /// Parses a <see cref="StackFrameParameterDeclarationNode"/> by parsing identifiers first representing the type and then the parameter identifier.
+        /// Ex: System.String[] s
+        ///     ^--------------^ -- Type = "System.String[]"
+        ///                     ^-- Identifier = "s"    
+        /// </summary>
+        private ParseResult<StackFrameParameterDeclarationNode> ParseParameterNode()
```

Comment:
required to return a result.  so call TryParseParameterNode and have caller abort if they get null back.